Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/LogExceptionAnalyzer.cs

```diff
@@ -35,13 +37,19 @@ public class LogExceptionAnalyzer : DiagnosticAnalyzer
 				description: Description
 			);
 
-		private string _logMethodNames;
-
+		private IEnumerable<string> _logMethodNames;
+
+		private const string InvalidSetupTitle = @"Log caught exceptions setup";
+		private const string InvalidSetupMessage = @"This analyzer requires an .editorconfig entry of the form dotnet_code_quality.{0}.{1} specifying a comma-separated list of allowed method calls inside catch blocks.";
+		private const string InvalidSetupDescription = @"This analyzer requires additional configuration in the .editorconfig.";
+		private static DiagnosticDescriptor InvalidSetupRule = new DiagnosticDescriptor(Helper.ToDiagnosticId(DiagnosticIds.LogException), InvalidSetupTitle, InvalidSetupMessage, Category, DiagnosticSeverity.Error, false, InvalidSetupDescription);

```

Comment:
I think you should provide a separate Diagnostic ID for this, as this is a different problem and required a different solution to fix.
Please note that developers (might) rely on the error code to find a fix (by Googling most of the time).
We could consider adding a general Diagnostic ID for missing configuration in .editorconfig, and use that ID in all the applicable Analyzers throughput the project.

Reply:
I agree, this should be a different PHxxxx number.  (I don't think we want a generic one).

Reply:
I think I disagree.  While I agree that in the code it's a different problem with a different solution, I don't think it's a different problem from a "customer perspective".  I.e., suppose the user wants to use this PH2090 analyzer.  This analyzer requires configuration that is very specific to PH2090.  If the help that is provided is done under the auspices of a different ID, then who is to say that that ID is enabled?  In that scenario, PH2090 will silently report nothing (because it's on but not configured, and PH20xx won't tell the user about PH2090 because it's not enabled.  The opposite is also true: what if they have PH20xx enabled but not PH2090.  In that scenario they would have an error telling them to go configure PH2090, even though it's not enabled.  Or maybe missing the spirit of this review comment?